src/components/plot/selectors-grid.tsx

```diff
@@ -0,0 +1,105 @@
+import DropdownSelect from '@isrd-isi-edu/deriva-webapps/src/components/plot/dropdown-select';
+import { Option } from '@isrd-isi-edu/deriva-webapps/src/components/virtualized-select';
+import { Responsive, WidthProvider } from 'react-grid-layout';
+import { gridLayoutConfigMap } from '@isrd-isi-edu/deriva-webapps/src/utils/string';
+import '/node_modules/react-resizable/css/styles.css';
+import '/node_modules/react-grid-layout/css/styles.css';
+import { GridLayoutConfig, SelectorConfig, LayoutConfig } from '@isrd-isi-edu/deriva-webapps/src/models/plot';
+import { PlotTemplateParams } from '@isrd-isi-edu/deriva-webapps/src/hooks/chart';
+
+
+type SelectorsGridProps = {
+    /**
+     * selectors data to be rendered
+     */
+    selectorData: {
+        gridConfig: GridLayoutConfig,
+        layout: LayoutConfig[],
+        selectorConfig: SelectorConfig[],
+        templateParams: PlotTemplateParams,
+    };
+    selectorOptions: Option[][];
+    setSelectorOptionChanged: any;
+    width: number | string;
+};
+
+//In simple cases a HOC WidthProvider can be used to automatically determine width upon initialization and window resize events.
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+
+/**
+ * It sets a new value in templateParams.$control_values based on selector, 
+ * triggers the setSelectorOptionChanged function, and returns the option
+ * @param option changed option
+ * @param selectorConfig configuration of the given selector
+ * @param templateParams 
+ * @param setSelectorOptionChanged setState method to indicate the change
+ * @returns 
+ */
+const handleChange = (option: Option, selectorConfig: SelectorConfig, templateParams: PlotTemplateParams, setSelectorOptionChanged: any) => {
+    if (option) {
+        setSelectorOptionChanged(true);
+        const uid = selectorConfig?.uid;
+        const valueKey = selectorConfig?.request_info?.value_key;
+        templateParams.$control_values[uid].values = {
+            [valueKey]: option.value, //else use default value
+        };
+        return option;
+    }
+};
+
+
+const SelectorsGrid = ({ selectorData, selectorOptions, setSelectorOptionChanged, width }: SelectorsGridProps): JSX.Element => {
+    const uid:string[]=[];
+    const valueKey:string[]=[];
+    const selectorValue: Option[]=[];
+    //Collect uid's and valueKey's for all selectors
+    selectorData.selectorConfig?.map((currentConfig,index)=>{
+        const currUid=currentConfig?.uid;
+        const currValueKey=currentConfig?.request_info.value_key;
+        uid.push(currUid);
+        valueKey.push(currValueKey);
+        const selectedOption=selectorOptions[index].find((option: Option) => 
```

Comment:
Is this looking through the existing data to initialize a selector? 

I think this makes sense since we should be reading query parameters and setting up `$control_values` if a query parameter is found.

Reply:
Yeah whatever is the value stored in the $control_values for the given selector should be selected value for that one.